edx/edx-enterprise
# [ENT-495] Create a unified Data Sharing Consent model

**Description:** This ticket creates a new `DataSharingConsent` model that represents the abstract concept of consent in preparation for use in future tickets, including [ENT-496](https://openedx.atlassian.net/browse/ENT-496).

**JIRA:** [ENT-495](https://openedx.atlassian.net/browse/ENT-495)

**Dependencies:** [ENT-496](https://openedx.atlassian.net/browse/ENT-496) depends on this and is in the same sprint.

**Merge deadline:** ASAP due to the dependency above.

**Testing instructions:**

1. See tests & test coverage for confidence.
2. See that the model(s) match the spec in the ticket.

There's not much else that can be done to test this because it's just adding models and nothing else -- it is [ENT-496](https://openedx.atlassian.net/browse/ENT-496) that will be using these models.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** I'm worried the `ProxyDataSharingConsent` "model" doesn't really match the spec, so I'll need @haikuginger to double check for any mistakes there.
